pkg/cmd/flags.go

```diff
@@ -96,17 +98,20 @@ func AddCloudFlag(cmd *cobra.Command) {
 
 func AddClusterFlag(cmd *cobra.Command, c *AuthenticatedCLICommand) {
 	cmd.Flags().String("cluster", "", "Kafka cluster ID.")
-	RegisterFlagCompletionFunc(cmd, "cluster", func(cmd *cobra.Command, args []string) []string {
-		if err := c.PersistentPreRunE(cmd, args); err != nil {
-			return nil
-		}
 
-		environmentId, err := c.Context.EnvironmentId()
-		if err != nil {
-			return nil
-		}
-		return AutocompleteClusters(environmentId, c.V2Client)
-	})
+	if c.V2Client != nil {
```

Comment:
Isn't it possible for this to be set in `PersistentPreRunE()`? If so, maybe we should wrap lines 108-112 with this `if`.